database/pgx/pgx.go

```diff
@@ -1,487 +1,32 @@
 //go:build go1.9
 // +build go1.9
 
+// Deprecated: Use github.com/golang-migrate/migrate/v4/database/pgx/v4 or
+// github.com/golang-migrate/migrate/v4/database/pgx/v5 instead, for pgx v4 and pgx v5 respectively.
 package pgx
 
 import (
-	"context"
-	"database/sql"
 	"fmt"
-	"io"
-	nurl "net/url"
-	"regexp"
-	"strconv"
-	"strings"
-	"time"
 
-	"go.uber.org/atomic"
-
-	"github.com/golang-migrate/migrate/v4"
-	"github.com/golang-migrate/migrate/v4/database"
-	"github.com/golang-migrate/migrate/v4/database/multistmt"
-	"github.com/hashicorp/go-multierror"
-	"github.com/jackc/pgconn"
-	"github.com/jackc/pgerrcode"
-	_ "github.com/jackc/pgx/v4/stdlib"
-)
-
-func init() {
-	db := Postgres{}
-	database.Register("pgx", &db)
-}
-
-var (
-	multiStmtDelimiter = []byte(";")
-
-	DefaultMigrationsTable       = "schema_migrations"
```

Comment:
These would need to be exported as well.
I think it's safer and easier to leave the default as v4 (e.g. remove the v4 folder in this PR) and update the README and godocs to also mention that a v5 driver is also available. I think it's fine to offer both v4 and v5 so we can mark v4 as deprecated later. e.g. whenever pgx deprecates v4. Do you know if v4 has already been deprecated?